# 4.33

## Encontrar el rango de los conjuntos: a) 5, 3, 8, 4, 7, 6, 12, 4, 3 y b) 8.772, 6.453, 10.624, 8.628, 9.434, 6.351.

In [1]:
a=[5, 3, 8, 4, 7, 6, 12, 4, 3]

In [2]:
a.sort()
max=a[-1]
min=a[0]

### Despues de ordenarlo de mayor a menor, tomamos el máximo menos el mínimo

In [3]:
max-min

9

### Respuesta a=9

In [4]:
b=[8.772, 6.453, 10.624, 8.628, 9.434, 6.351]
b.sort()
max=b[-1]
min=b[0]

In [5]:
max-min

4.273000000000001

### Respuesta 4.273

### Análogamente obtenemos 4.273 

# 4.37
## En la tabla siguiente se dan las semanas que necesitaron 25 trabajadores, que perdieron su trabajo por reducción de perso- nal en sus empresas, para encontrar un nuevo empleo. Encontrar el rango de estos datos.

In [6]:
tabla=[13,13,17,7,22,22,26,17, 13,14,16,7,6,18,20,10,17,11,10,15,16, 8, 16, 21, 11]

In [7]:
tabla.sort()
max=tabla[-1]
min=tabla[0]

In [8]:
max-min

20

### Respuesta 20

# 4.39

## 4.39 Encontrar la desviación media de los conjuntos: a) 3, 7, 9, 5 y b) 2.4, 1.6, 3.8, 4.1, 3.4.

In [9]:
a =[3, 7, 9, 5] 
b =[2.4, 1.6, 3.8, 4.1, 3.4]

In [10]:
import pandas as pd

def desviacion_media(lista: list) -> float:
    dfa=pd.DataFrame(lista)
    media =dfa[0].mean()

    suma=0
    for element in a:
        suma += abs(element - media)
    return suma/len(lista)

In [11]:
desviacion_media(a)

2.0

In [12]:
desviacion_media(b)

2.3760000000000003

## a = 2  y b = 2.376

# 4.49 

## Encontrar el rango semiintercuartil de los datos dados en el problema 4.37

In [13]:
tabla=[13,13,17,7,22,22,26,17, 13,14,16,7,6,18,20,10,17,11,10,15,16, 8, 16, 21, 11]

In [14]:
df=pd.DataFrame(tabla)
df[0].describe()

count    25.000000
mean     14.640000
std       5.219195
min       6.000000
25%      11.000000
50%      15.000000
75%      17.000000
max      26.000000
Name: 0, dtype: float64

$$Q = \frac{Q_3 - Q_1}{2}$$

In [15]:
Q = (17 - 11) /2
Q

3.0

## Respuesta 3

# 4.52 
### Encontrar el rango percentil 10-90 en las distribuciones: a) del problema 3.59 y b) del problema 3.107. En cada caso interpretar los resultados claramente.

In [16]:
df=pd.read_csv("359.csv")
df.rename(columns={"Unnamed: 0": "carga", "0":"Frecuencias (f)"},inplace=True)
df

carga  Frecuencias (f)
0    9.3-9.7                2
1   9.8-10.2                5
2  10.3-10.7               12
3  10.8-11.2               17
4  11.3-11.7               14
5  11.8-12.2                6
6  12.3-12.7                3
7  12.8-13.2                1

In [17]:
df['F'] = df['Frecuencias (f)'].cumsum()
df

carga  Frecuencias (f)   F
0    9.3-9.7                2   2
1   9.8-10.2                5   7
2  10.3-10.7               12  19
3  10.8-11.2               17  36
4  11.3-11.7               14  50
5  11.8-12.2                6  56
6  12.3-12.7                3  59
7  12.8-13.2                1  60

In [18]:
def descubrir_intervalo_de__clase(
    df: pd.DataFrame, clase: str = "Clase", separador: str = "-"
):
    """Funcion para encontrar el intervalo de clase de manera sencilla"""
    superior = (
        float(df[clase][1].split(separador)[0])
        + float(df[clase][0].split(separador)[1])
    ) / 2
    inferior = (
        float(df[clase][2].split(separador)[0])
        + float(df[clase][1].split(separador)[1])
    ) / 2
    intervalo_de_clase = inferior - superior
    return intervalo_de_clase

$$P_k = L_i + \frac{\frac{k N}{100}-F_{i-1}}{f_i}a_i   $$

10 percentil

In [19]:
N=df["Frecuencias (f)"].sum()
k=10
k*N/100

6.0

El décimo percentil debe de estar en el segundo grupo 9.8-10.2

In [20]:
L = 9.75
F_menos_1 = 2
f = 5

In [21]:
a_i= descubrir_intervalo_de__clase(df,"carga")
a_i

0.5

In [22]:
P_10 = L + (((k*N/100)-F_menos_1)/f)*a_i

In [23]:
P_10

10.15

### Ahora P_90

In [24]:
N=df["Frecuencias (f)"].sum()
k=90
k*N/100

54.0

El 90-vo percentil deve de estar en el grupo 11.8-12.2	

In [25]:
L = 11.75
F_menos_1 = 50
f = 6

In [26]:
P_90 = L + (((k*N/100)-F_menos_1)/f)*a_i
P_90

12.083333333333334

## 10% de la carga  está debajo de 10.5 y 90% de la carga está debajo de 12.08

# P_90 = 38.6 y P_10 = 12.6

### inciso B

In [27]:
df=pd.read_csv("3107.csv")
df.rename(columns={"Unnamed: 0": "Calificación", "0":"Estudiantes"},inplace=True)
df

Calificación  Estudiantes
0       90-100            9
1        80-89           32
2        70-79           43
3        60-69           21
4        50-59           11
5        40-49            3
6        30-39            1

In [28]:
df['F'] = df['Estudiantes'].cumsum()
df

Calificación  Estudiantes    F
0       90-100            9    9
1        80-89           32   41
2        70-79           43   84
3        60-69           21  105
4        50-59           11  116
5        40-49            3  119
6        30-39            1  120

$$P_k = L_i + \frac{\frac{k N}{100}-F_{i-1}}{f_i}a_i   $$

In [29]:
N=df["Estudiantes"].sum()
k=10
k*N/100

12.0

Entonces P_10 debe estar en la clase 80-89

In [30]:
L = 89.5
F_menos_1 = 9
f = 32

In [31]:
a_i= descubrir_intervalo_de__clase(df,"Calificación")
a_i

-10.5

In [32]:
P_10 = L + (((k*N/100)-F_menos_1)/f)*a_i
P_10

88.515625

### Respuesta, solo el 10% está arriba de 88.5%
### Ahora P_90

In [33]:
N=df["Estudiantes"].sum()
k=90
k*N/100

108.0

El intervalo debe estar entre 50-59

In [34]:
L = 59.4
F_menos_1 = 105
f = 11

In [35]:
P_90 = L + (((k*N/100)-F_menos_1)/f)*a_i
P_90

56.53636363636363

### 90 % de los estudiantes están arriba de 56.5%

# 4.56
### Encontrar la desviación estándar de los conjuntos: a) 3, 6, 2, 1, 7, 5; b) 3.2, 4.6, 2.8, 5.2, 4.4, y c) 0, 0, 0, 0, 0, 1, 1, 1

In [36]:
a = [3, 6, 2, 1, 7, 5]
dfa=pd.DataFrame(a)
dfa.describe().loc["std"][0]

2.3664319132398464

Respuesta 2.36

In [37]:
b = [3.2, 4.6, 2.8, 5.2, 4.4]
dfb=pd.DataFrame(b)
dfb.describe().loc["std"][0]

1.0039920318408908

Respuesta 1.003992

In [38]:
c = [0, 0, 0, 0, 0, 1, 1]
dfc=pd.DataFrame(c)
dfc.describe().loc["std"][0]

0.48795003647426655

Respuesta 0.48795